City-Bureau/city-scrapers
# 🕷️ Fix spider: Illinois Criminal Justice Information Authority

## What's this PR do?

Fixes our Illinois Criminal Justice Information Authority spider (aka. `il_criminal_justice_information`).

## Why are we doing this?

The spider broke due to changes on the pages it's targeting. The changes in this PR ensure the scraper runs without error.

## Steps to manually test

After installing the project using `pipenv`:

1. Activate the virtual environment:
```
pipenv shell
```
2. Run the spider:
```
scrapy crawl il_criminal_justice_information -O test_output.csv
```
3. Monitor the stdout and ensure that the crawl proceeds without raising any errors. Pay attention to the final status report from scrapy.

4. Inspect `test_output.csv` to ensure the data looks valid. I suggest opening a few of the URLs under the source column of test_output.csv and comparing the data for the row with what you see on the page.

## Are there any smells or added technical debt to note?
- This rebuilt spider request all meetings from the agency's GraphQL endpoint. It's possible the request could be optimized so we filter out meetings older than 90 days in the query but I haven't toyed with the agency's API enough to determine if that's possible. The current approach seemed decent enough.
